pywhy_graphs/classes/diungraph.py

```diff
@@ -184,6 +143,90 @@ def possible_parents(self, n: Node) -> Iterator[Node]:
         """
         return self.sub_undirected_graph().neighbors(n)
 
+
+class CPDAG(DiUnGraph, ConservativeMixin):
+    """Completed partially directed acyclic graphs (CPDAG).
```

Comment:
Perhaps it is worth adding a note or warning even to the classes that validity is not guaranteed. Please use the validity check if you need to rigorously check that your construction is valid?

Same in chaingraph? Maybe also in the ADMG, and PAG too?

Reply:
Yeah this is a good idea. I will address just CPDAG and Chain Graph in this PR but will look to redo the other classes in another PR.